### DISCUSSION TITLE
Anyone working on latency?

### DISCUSSION THREAD

**Discussion by arahman33** [omnetpp/omnetpp#1151] [Q&A]
Hi everyone,

I am trying to get the latencies through each link. However, the statistics only (to the best of my knowledge) offer mean packet lifetime which can be interpreted as the mean latency. In order to get the mean, it should somewhere gather all the latencies. Does anyone know where to find that?

Any direction/suggestion would be highly appreciated. Thanks!

**Comment by adamgeorge309** [comment]
The statistic `meanBitLifeTimePerPacket` is a vector, and contains the latency for each packet (it is a mean of the delays of all the bits of the packet). So this might be the one you want.

**Reply by arahman33** [reply]
I think I might have been understanding that statistics entirely wrong. It's not the mean delay of the packet, rather the mean of the bits in the packet, which essentially means the delay of the entire packet? 
How if the received packets were from different source nodes but the receiver is one. How would this work?

**Reply by levy** [reply]
Most often all bits in a packet has the same end-to-end delay because they take the same path together. Sometimes it's not the case. For example in a TCP stream, the source may generate 100B at a time and TCP segments may be 536B large. Therefore some bytes of a TCP segment in the receiver are older than others.
